promoted_python_delivery_client/client/client.py

```diff
@@ -0,0 +1,199 @@
+import concurrent.futures
+import copy
+import logging
+from typing import List, Optional
+import time
+import uuid
+from promoted_python_delivery_client.client.api_delivery import APIDelivery
+from promoted_python_delivery_client.client.api_metrics import APIMetrics
+from promoted_python_delivery_client.client.delivery_request import DeliveryRequest
+from promoted_python_delivery_client.client.delivery_response import DeliveryResponse
+from promoted_python_delivery_client.client.sampler import Sampler
+from promoted_python_delivery_client.client.sdk_delivery import SDKDelivery
+from promoted_python_delivery_client.model.client_info import ClientInfo
+from promoted_python_delivery_client.model.client_type import ClientType
+from promoted_python_delivery_client.model.cohort_arm import CohortArm
+from promoted_python_delivery_client.model.cohort_membership import CohortMembership
+from promoted_python_delivery_client.model.delivery_execution import DeliveryExecution
+from promoted_python_delivery_client.model.delivery_log import DeliveryLog
+from promoted_python_delivery_client.model.execution_server import ExecutionServer
+from promoted_python_delivery_client.model.log_request import LogRequest
+from promoted_python_delivery_client.model.request import Request
+from promoted_python_delivery_client.model.response import Response
+from promoted_python_delivery_client.model.timing import Timing
+from promoted_python_delivery_client.model.traffic_type import TrafficType
+
+
+SERVER_VERSION = "python.1.0.0"
+
+# Executor to run metrics logging in the background.
+DEFAULT_METRICS_THREAD_POOL_SIZE = 5
+
+# Default number of maximum request insertion passed to Delivery API.
+DEFAULT_MAX_REQUEST_INSERTIONS = 1000
+
+
+class PromotedDeliveryClient:
+    def __init__(self,
+                 delivery_endpoint: str,
+                 delivery_api_key: str,
+                 metrics_endpoint: str,
+                 metrics_api_key: str,
+                 delivery_timeout_millis: Optional[int] = None,
+                 metrics_timeout_millis: Optional[int] = None,
+                 max_request_insertions: Optional[int] = DEFAULT_MAX_REQUEST_INSERTIONS,
+                 shadow_traffic_delivery_rate: Optional[float] = 0,
+                 perform_checks: Optional[bool] = False) -> None:
+        self.metrics_endpoint = metrics_endpoint
+        self.metrics_api_key = metrics_api_key
+        self.metrics_timeout_millis = metrics_timeout_millis
+        self.max_request_insertions = max_request_insertions
+        self.sampler = Sampler()
+        self.sdk_delivery = SDKDelivery()
+        self.api_delivery = APIDelivery(delivery_endpoint, delivery_api_key, delivery_timeout_millis)
+        self.api_metrics = APIMetrics(metrics_endpoint, metrics_api_key, metrics_timeout_millis)
+
+        if shadow_traffic_delivery_rate is not None and (shadow_traffic_delivery_rate < 0 or shadow_traffic_delivery_rate > 1):
```

Comment:
Since this is optional is it the case that shadow_traffic_delivery_rate 0 and None are treated equivalently?

If there's a difference between meaning of 0 and None it might be worth documenting. Alternatively, if there's some implicit default that "None" means it might make sense to change the parameter default from None to whatever the eventual default this gets set to is.

If 0 and None are equivalent--this is what I assume as a reader--then it might be more straightforward to just check
```
if shadow_traffic_delivery_rate and (shadow_traffic_delivery_rate < 0 or shadow_traffic_delivery_rate > 1)
```
remove thing "is not None" since this implies to me that you want to distinguish between None and 0 values.

Reply:
Per your other comment, I made it non-Optional-default. Thanks!